src/allmydata/client.py

```diff
@@ -1125,8 +1125,18 @@ def create_node_from_uri(self, write_uri, read_uri=None, deep_immutable=False, n
         # may get an opaque node if there were any problems.
         return self.nodemaker.create_from_cap(write_uri, read_uri, deep_immutable=deep_immutable, name=name)
 
-    def create_dirnode(self, initial_children=None, version=None):
-        d = self.nodemaker.create_new_mutable_directory(initial_children, version=version)
+    def create_dirnode(
+        self,
+        initial_children=None,
+        version=None,
+        *,
```

Comment:
I don't think we use this syntax or type-hinting anywhere else in this codebase, so might cause more confusion than it solves? Ideally should gain a docstring (too/instead).

Reply:
I've added a docstring via https://github.com/tahoe-lafs/tahoe-lafs/pull/1365/commits/f694224aba0adec125786097d744a694fc72ec0f

The syntax and type-hinting follow the already-existing `create_mutable_file` function (just two functions down in the same module) which I was borrowing from. https://github.com/tahoe-lafs/tahoe-lafs/pull/1365/commits/00f82a46ae5fae5a706c75fc204a1f002dc13bce adds the missing hints (if that's what you were referring to?) but I can take them out entirely and/or inline everything if that's what you would prefer -- just let me know. (I found a number of conflicting "styles" in use throughout the codebase).

Reply:
thanks for the docstring. leaving the type-hints is fine / good (just AFAIK they won't really "do" anything but that can come later)